pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -194,54 +201,67 @@ func (configgen *ConfigGeneratorImpl) buildSidecarListeners(env *model.Environme
 	actualWildcard, actualLocalHostAddress := getActualWildcardAndLocalHost(node)
 
 	if mesh.ProxyListenPort > 0 {
-		inbound := configgen.buildSidecarInboundListeners(env, node, push, proxyInstances)
-		outbound := configgen.buildSidecarOutboundListeners(env, node, push, proxyInstances)
+		// Notes that the the else branch works for both split and non-split cases.
+		// TODO(silentdai): remove `if` branch once split behavior is well verified
+		if !node.IsInboundCaptureAllPorts() {
```

Comment:
?? This is a lot of code duplication, not to mention double maintenance. I really liked the object chaining you had. This file is big as is, so lets avoid adding more duplicate code paths for no good.
If stability is a concern, split the object chaining thing as a separate PR and let that merge as a standard code consolidation/refactor PR. Then this would simply be adding an if-else loop.

Reply:
IMO stability is the top concern. 
Current code is already too complex and doesn't yet have full coverage - so any refactoring is dangerous, in particular this close to a release and for code that is not yet clear will stay that way. 
Easy to remove unused code - once we prove the new code works - and the remaining code will be cleaner.

Reply:
The context is that we wanna provide this opt-in feature at 1.2 which kicks off several days away, 05/20 IIRC

I don't have enough time to fully migrate the code to the builder pattern before that date. The main blocker is the test case.

So the follow up would be:

- [x] borrow existing `ConfigGenerator`. The builder mostly invoke the `ConfigGenerator` function
- [x] When the feature is off, the flow is following the existing code path literally. The test cases of the new builder only cover new feature(although it is supposed to cover both on and off) 
- [] migrate the new builder to use filter chain only at inbound listener
- [] Adding more and more test cases to verify the builder build when the feature is off
- [] migrate `ConfigGenerator` code to the `Builder`

Reply:
The context is that this will be an alpha feature - while the code we have in place is more or less stable. We don't have enough test coverage to be able to refactor things freely.